### PR TITLE
Add Vote pallet 

### PR DESCRIPTION
**Pull request by benoitdevos**
* Given that (s)he is owner of a Closed, non-Void LOC,
a Legal Officer can create a Vote associated with this LOC.
* Voters (in this case all LLOs) are determined at vote creation time.

https://github.com/logion-network/logion-internal/issues/713

### CODE REVIEW

File: pallet-logion-vote/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+[package]
+authors = ['Logion Team <https://github.com/logion-network>']
+description = 'Pallet implements Logion Vote.'
+edition = '2021'
+homepage = 'https://logion.network'
+license = 'Apache-2.0'
+name = 'pallet-logion-vote'
+readme = 'README.md'
+version = '0.1.1'
```
Comment (gdethier) [inline pallet-logion-vote/Cargo.toml:9]: 0.1.0

File: pallet-logion-vote/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo)]
+pub struct Ballot<AccountId> {
+    voter: AccountId,
+    status: BallotStatus,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo)]
+pub enum BallotStatus {
+    UnVoted,
```
Comment (gdethier) [inline pallet-logion-vote/src/lib.rs:29]: To me, un-voted means that you voted before and you "cancelled" your vote. Maybe use "not voted" instead?

File: pallet-logion-vote/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub (super) trait Store)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    /// Votes
+    #[pallet::storage]
+    #[pallet::getter(fn votes)]
+    pub type Votes<T> = StorageValue<_, Vec<Vote<<T as Config>::LocId, <T as frame_system::Config>::AccountId>>, ValueQuery>;
```
Comment (gdethier) [inline pallet-logion-vote/src/lib.rs:68]: As already discussed, switch to a map and add an additional storage value for the counter.

File: pallet-logion-vote/src/mock.rs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+impl LocValidity<<Test as pallet_logion_vote::Config>::LocId, <Test as system::Config>::AccountId> for LocValidityMock {
+    fn loc_valid_with_owner(loc_id: &<Test as pallet_logion_vote::Config>::LocId, legal_officer: &<Test as Config>::AccountId) -> bool {
+        return *loc_id == LOC_ID && *legal_officer == LEGAL_OFFICER1;
+    }
+}
+
+parameter_types! {
+	pub const BlockHashCount: u64 = 250;
+	pub const SS58Prefix: u8 = 42;
```
Comment (gdethier) [inline pallet-logion-vote/src/mock.rs:54]: Wrong indent

File: pallet-logion-vote/src/lib.rs (hunk 2 above)
Comment (benoitdevos) [reply pallet-logion-vote/src/lib.rs:29]: Agreed !

File: pallet-logion-vote/src/lib.rs (hunk 3 above)
Comment (benoitdevos) [reply pallet-logion-vote/src/lib.rs]: Done.

**Review by gdethier — APPROVED** [review]
Great!

**Pull request merged**